File: base_framework/src/main/java/siarhei/luskanau/android/framework/interactor/SchedulerSet.java
package siarhei.luskanau.android.framework.interactor;

import android.support.annotation.NonNull;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class SchedulerSet implements ISchedulerSet {

    @NonNull
    @Override
    public Scheduler subscribeOn() {
        return Schedulers.io();
    }

    @NonNull
    @Override
    public Scheduler observeOn() {
        return AndroidSchedulers.mainThread();
    }
}

File: app/src/main/java/siarhei/luskanau/iot/doorbell/data/SchedulerSet.kt
package siarhei.luskanau.iot.doorbell.data

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import io.reactivex.schedulers.TestScheduler

class SchedulerSet(
        val io: Scheduler = Schedulers.io(),
        val ui: Scheduler = AndroidSchedulers.mainThread(),
        val computation: Scheduler = Schedulers.computation()
) {

    companion object {
        fun test(): SchedulerSet = SchedulerSet(
                TestScheduler(),
                TestScheduler(),
                TestScheduler()
        )
    }

    fun triggerActions() {
        if (io is TestScheduler) {
            io.triggerActions()
        }
        if (ui is TestScheduler) {
            ui.triggerActions()
        }
        if (computation is TestScheduler) {
            computation.triggerActions()
        }
    }

}